src/screens/Sensors/index.test.tsx

```diff
@@ -0,0 +1,160 @@
+import SensorModel from '../../database/model/sensorModel';
+import {Sensor} from '../../components/Sensor';
+
+let screen!: ReactTestInstance;
+let tree!: ReactTestRenderer;
+let records: SensorModel[] = [];
+
+beforeEach(async () => {
+  // becuase this screen relies
+  // on providers like globalData
+  // navigation etc. We'll just load
+  // the app then navigate to it.
+  await renderer.act(async () => {
+    tree = renderer.create(<App />);
+  });
+
+  renderer.act(() => {
+    navigate('Sensors');
+  });
+
+  screen = tree.root.findByType(Sensors);
+});
+
+afterEach(async () => {
+  await dataBase.write(async () => {
+    for (let record of records) {
+      await record.destroyPermanently();
+    }
+  });
+
+  tree.unmount();
+});
+
+it('Screen renders correctly when there ARE NO sensors', async () => {
+  // check that it renders when a new item is added to the db.
+  const text = screen.findByType(Text);
+  expect(text.props.children).toBe('No Connected Sensors yet... Try add some!');
+});
+
+it('Screen renders correctly when there ARE sensors', async () => {
+  // check that it renders when a new item is added to the db.
+  const list = screen.findByType(FlatList);
+  const sensorName = 'ble-1234';
+
+  const sensors = screen.findAllByType(Sensor);
+  expect(sensors.length).toBe(0);
+
+  await renderer.act(async () => {
+    const record = await createSensor(sensorName, 'xyz', []);
+    records.push(record);
+    list.props.onRefresh();
+  });
+
+  const sensor = screen.findByType(Sensor);
+
+  expect(sensor.props.data.name).toBe(sensorName);
+});
+
+it('Screen allows removing a sensor', async () => {
+  const list = screen.findByType(FlatList);
+  const sensorName = 'ble-1234';
+
+  // add a sensor
+  await renderer.act(async () => {
+    const record = await createSensor(sensorName, 'xyz', []);
+    records.push(record);
+    list.props.onRefresh();
+  });
+
+  // check it's there
+  const sensor = screen.findByType(Sensor);
+  expect(sensor.props.data.name).toBe(sensorName);
+
+  // remove sensor + refresh
+  await renderer.act(async () => {
+    await sensor.props.onAction();
+    list.props.onRefresh();
+  });
+
+  // check it's gone
+  const sensors = screen.findAllByType(Sensor);
+  expect(sensors.length).toBe(0);
+});
+
+it('Screen allows adding a sensor', async () => {
+  // 1. click add bring up modal
+  // 2. press scan
+  // 3. pair item from discovery list
+  // 4. check sensor is saved to db
+  // 5. and is loaded on original screen.
+  const sensorName = 'a-heart-rate-monitor';
+  const addButton = tree.root.findByType(Button);
+
+  await renderer.act(async () => {
+    await addButton.props.onPress();
+  });
+
+  // look for empty list text
+  const text = tree.root
+    .findAllByType(Text)
+    .find(t => t.props.children === 'No sensors found, run a scan!');
+  expect(text).toBeTruthy();
+
+  const scanButton = tree.root.findByProps({id: 'discovery-modal-scan'});
```

Comment:
@shaunmulligan I wasn't sure what a real sensor payload would look like, happy to change the `sensorsDiscovered` test data to make it more realistic.

Reply:
So the discovered sensors would be in an array that looks like:
```
 [{"advertising": {"isConnectable": true, "localName": "Forerunner 945", "manufacturerData": [Object], "serviceData": [Object], "serviceUUIDs": [Array], "txPowerLevel": -2147483648}, "id": "F0:99:19:59:B4:00", "name": "Forerunner 945", "rssi": -69, "sensorType": ["HeartRate"]}]
```
and a paired sensor would look like in the DB. We might want to make them more similar, so they are easier to work with.
<img width="1173" alt="Screenshot 2023-03-22 at 07 50 45" src="https://user-images.githubusercontent.com/3915043/226824381-0c1cc346-da22-4a44-8058-4a023c64a7c0.png">